### PR TITLE
Bump version of our pre-commit hooks

### PR DESCRIPTION
**Pull request by jonathangreen**
## Description

This one touches a lot of files, but it shouldn't be changing any behaviour. Everything in this PR was automatically generated by pre-commit.

All I did to create this PR was:
```
pre-commit autoupdate
pre-commit run -a
```

## Motivation and Context

While working on PP-1769 I ran into an issue with the `black` pre-commit hook:
```
error: cannot format src/palace/manager/opds/base.py: Cannot parse: 102:12:             match (rel, type):
```

It looks like the older version of `black` we were using couldn't handle that match statement, I could have tried to work around it, but I figured I might as well just update the version we were using.

## How Has This Been Tested?

- Tests in CI

## Checklist

- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.

### CODE REVIEW

File: bin/informational/adobe-id-for-short-client-token [hunk 1]
```diff
@@ -23,8 +23,7 @@ REGISTRY_RESPONSE_RE = re.compile(
 )
 
 
-class InvalidTokenException(Exception):
-    ...
+class InvalidTokenException(Exception): ...
```
Comment (tdilauro) [inline bin/informational/adobe-id-for-short-client-token:26]: Yay!

File: src/palace/manager/api/axis.py [hunk 2]
```diff
@@ -262,7 +262,7 @@ def authorization_headers(self) -> dict[str, str]:
         )
         return dict(Authorization="Basic " + authorization_b64)
 
-    def _run_self_tests(self, _db: Session) -> Generator[SelfTestResult, None, None]:
+    def _run_self_tests(self, _db: Session) -> Generator[SelfTestResult]:
```
Comment (tdilauro) [inline src/palace/manager/api/axis.py:265]: Type defaults! Assuming this came from pyupgrade.

**Review by tdilauro — APPROVED** [review]
This looks fine to me. 🏆 

PEP 696 type defaults FTW!

File: src/palace/manager/api/axis.py (hunk 2 above)
Comment (jonathangreen) [reply src/palace/manager/api/axis.py:265]: It did! I was happy to see this as well!

**Pull request merged**